mfr/core/extension.py

```diff
@@ -1,11 +1,25 @@
 import abc
+import asyncio
 
+import uuid
+
+import waterbutler
+from waterbutler.core.streams import StringStream
+
+from mfr.core import utils
 from mfr.core.metrics import MetricsRecord
+from mfr.server import settings
+
+
+class Cacheable(metaclass=abc.ABCMeta):
 
+    @abc.abstractproperty
+    def cache_result(self):
+        pass
 
 class BaseExporter(metaclass=abc.ABCMeta):
 
-    def __init__(self, ext, source_file_path, output_file_path, format):
+    def __init__(self, metadata, input_stream, format):
```

Comment:
pass the stream in to the exporter so the subcalsses' export method can choose to use the stream rather than writing to disk if it is feasable and appropriate.